project_manager.py

```diff
@@ -265,6 +292,57 @@ def _load_sublime_project_files(self, folder):
                     os.rmdir(d)
         return pfiles
 
+    def _auto_discover_repos(self):
+        try:
+            user_discover_dirs = pm_settings.get('auto_discover_base_directories')
+
+            # Get a list of discover_dirs to search for projects
+            node = computer_name()
+            if isinstance(user_discover_dirs, dict):
+                if node in user_discover_dirs:
+                    user_discover_dirs = user_discover_dirs[node]
+                else:
+                    user_discover_dirs = []
+            if isinstance(user_discover_dirs, str):
+                user_discover_dirs = [user_discover_dirs]
+
+            discover_dirs = []
+            for folder in user_discover_dirs:
+                p = expand_path(folder)
+                p = p.replace("$hostname", node)
+                discover_dirs.append(p)
+
+            # Convert existing project info to dict for easy lookup
+            existing = {info['folder']:key for key,info in self._info.items()}
+
+            # Find new repo directories
+            max_depth = pm_settings.get('auto_discover_max_recursion_depth', 1)
+            ignores = pm_settings.get('auto_discover_ignored_folders', [])
+            for basedir in discover_dirs:
+                if not os.path.isdir(basedir):
+                    print('ProjectManager: {} is not a valid path'.format(basedir))
+                for folder, repodir in find_repos(basedir, max_depth, ignores):
+                    if folder not in existing:
+                        project_name = os.path.basename(folder)
+                        self._create_new_project(project_name, folder)
+        except Exception as err:
+            print(err)
+
+    def _create_new_project(self, project_name, folder):
+        data = {"folders": [{
+            "binary_file_patterns": [],
+            "file_exclude_patterns": [],
+            "folder_exclude_patterns": [],
+            "name": project_name,
+            "path": folder,
+        }]}
+        
+        filename = '{}.sublime-project'.format(project_name)
+        filepath = os.path.join(self.primary_dir(), filename)
+        print('ProjectManager: Creating {} -> {}'.format(filename, folder))
+        with open(filepath, 'w') as handle:
```

Comment:
You might use `JsonFile(path).save(data)`

Reply:
Good catch. Updated to use JsonFile and removed the import above.